### DISCUSSION TITLE
BLE Integration - 6 BLE Tags configured - only 1 goes across MQTT to Home Assistant

### DISCUSSION THREAD

**Discussion by bleitchhk** [mKeRix/room-assistant#454] [Q&A]
Hi;

I am using the BLE Integration with 6 Blue Charm BLE Tags.  Only 1 seems to register as a device via the MQTT broker.

I am using integrated Mosquito in Home Assistant, HASSIO image on Pi 4b.

using 6 hosts on Pi 0Ws on same subnet of 192.168.11/24 (single subnet, not mdns issues it hope).  Broker also on same subnet.

Problem
====== 
- Configuring 6 BLE tags - same UUID, Same Major, incrementing minor
- All instances of Room Assistant connect to MQTT Broker
- All instances of Room Assistant successfully recognise all BLE tags in manual room-assistant startup via SSH
- only 1 of the 6 BLE tags successfully register sensors via MQTT and by definition only 1 device tracker
-    the one that works, works fully 
- 1 working below is Solo Altus Keys 1 All

Could someone please help me understand why only 1 tag will work ?

The one tag that is working is working very well through the whole system.

Mosquito logs - shows hosts connecting
============================== 
Addresses 30-35 are my Room Assistant PI addresses

1610817323: New connection from 192.168.11.32 on port 1883.
1610817323: New client connected from 192.168.11.32 as mqttjs_d85332ea (p2, c1, k60, u'mqttsabuca').
1610817355: New connection from 192.168.11.35 on port 1883.
1610817355: New client connected from 192.168.11.35 as mqttjs_271a5dac (p2, c1, k60, u'mqttsabuca').
1610817414: Socket error on client mqttjs_271a5dac, disconnecting.
1610817445: Client mqttjs_30355d15 disconnected.
1610817483: New connection from 192.168.11.34 on port 1883.
1610817483: New client connected from 192.168.11.34 as mqttjs_482d5761 (p2, c1, k60, u'mqttsabuca').
1610817491: New connection from 192.168.11.35 on port 1883.
[INFO] found mqttsabuca on Home Assistant
1610817491: New client connected from 192.168.11.35 as mqttjs_5657c1c7 (p2, c1, k60, u'mqttsabuca').
1610817571: Socket error on client mqttjs_482d5761, disconnecting.
1610817647: New connection from 192.168.11.34 on port 1883.
1610817647: New client connected from 192.168.11.34 as mqttjs_66482e5e (p2, c1, k60, u'mqttsabuca').
1610817989: New connection from 192.168.11.59 on port 1883.
[INFO] found mqttsabuca on Home Assistant
1610817990: New client connected from 192.168.11.59 as shellyht-2E2BFE (p2, c1, k60, u'mqttsabuca').
1610818081: Client shellyht-2E2BFE has exceeded timeout, disconnecting.
1610818095: New connection from 192.168.11.52 on port 1883.
1610818095: New client connected from 192.168.11.52 as shellyht-2D5D10 (p2, c1, k60, u'mqttsabuca').
1610818122: Socket error on client shellyht-2D5D10, disconnecting.
1610818601: Client mqttjs_63860c1b disconnected.
1610818638: New connection from 192.168.11.31 on port 1883.
[INFO] found mqttsabuca on Home Assistant
1610818638: New client connected from 192.168.11.31 as mqttjs_0662f968 (p2, c1, k60, u'mqttsabuca').
1610818889: New connection from 192.168.11.59 on port 1883.
1610818889: New client connected from 192.168.11.59 as shellyht-2E2BFE (p2, c1, k60, u'mqttsabuca').
1610818905: Socket error on client mqttjs_0662f968, disconnecting.
1610818980: Client shellyht-2E2BFE has exceeded timeout, disconnecting.
1610819006: New connection from 192.168.11.31 on port 1883.
[INFO] found mqttsabuca on Home Assistant

Room Assistant devices / entities appearing under MQTT integration under Home Assistant:
===================================================================
entry-room-assist-pi0w-6ab0ff-30
Raspberry Pi Foundation
Raspberry Pi Zero W Rev 1.1
No area
MQTT
-
guest-room-assist-pi0w-e5c4f1-34
Raspberry Pi Foundation
Raspberry Pi Zero W Rev 1.1
No area
MQTT
-

kitchen-room-assist-pi0w-131e32-31
Raspberry Pi Foundation
Raspberry Pi Zero W Rev 1.1
No area
MQTT
-
living-room-assist-pi0w-c77ddb-33
Raspberry Pi Foundation
Raspberry Pi Zero W Rev 1.1
No area
MQTT
-
master-room-assist-pi0w-593788-35
Raspberry Pi Foundation
Raspberry Pi Zero W Rev 1.1
No area
MQTT
-
SoloAltusKeys1All
<unknown>
<unknown>
No area
MQTT
-
tvroom-room-assist-pi0w-5e7113-32
Raspberry Pi Foundation
Raspberry Pi Zero W Rev 1.1
No area
MQTT
-
local.yml full config
============== 
global:
  integrations:
    - homeAssistant
    - bluetoothLowEnergy
#
homeAssistant:
  mqttUrl: 'mqtt://192.168.11.2:1883'
  mqttOptions:
    username: xxxxxxx
    password: xxxxxxx
#
bluetoothLowEnergy:
  whitelist:
    - 42c67565436861726d426561636f6e73-3838-4950
    - 426c7565436861726d426561636f6e73-3838-4949 (only this one works -> SoloAltusKeys1All)
         NOTE:  the working one above was originally configured 1st and ONLY, I moved it around to see if I could get another tag to work.
    - 42c67565436861726d426561636f6e73-3838-4951
    - 42c67565436861726d426561636f6e73-3838-4952
    - 42c67565436861726d426561636f6e73-3838-4953
    - 42c67565436861726d426561636f6e73-3838-4954

Device Tracker Config in Home Assistant:
=============================== 
    solo_altus_keys_1_all_bletag: 'room-assistant/device_tracker/ble-426c7565436861726d426561636f6e73-3838-4949-tracker/state'
    solo_altus_keys_2_mini_bletag: 'room-assistant/device_tracker/ble-426c7565436861726d426561636f6e73-3838-4950-tracker/state'
    solo_altus_keys_3_parents_bletag: 'room-assistant/device_tracker/ble-426c7565436861726d426561636f6e73-3838-4951-tracker/state'
    solo_altus_keys_4_spare_bletag: 'room-assistant/device_tracker/ble-426c7565436861726d426561636f6e73-3838-4952-tracker/state'
    solo_altus_keys_5_concierge_bletag: 'room-assistant/device_tracker/ble-426c7565436861726d426561636f6e73-3838-4953-tracker/state'
    solo_altus_keys_6_patti_bletag: 'room-assistant/device_tracker/ble-426c7565436861726d426561636f6e73-3838-4954-tracker/state'

Logs from a startup of one of the Pi 0Ws
============================== 
"SoloAltusKeys" named entities are the device names in beacons.  Room Assistant is discovering the devices.

My tags seem to be picket up ?
pi@kitchen-room-assist-pi0w-131e32-31:~ $ sudo systemctl stop room-assistant
pi@kitchen-room-assist-pi0w-131e32-31:~ $ cd ~/room-assistant/
pi@kitchen-room-assist-pi0w-131e32-31:~/room-assistant $ room-assistant
*** WARNING *** The program 'node' uses the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
*** WARNING *** The program 'node' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
1/16/2021, 6:26:42 PM - info - IntegrationsModule: Loading integrations: home-assistant, bluetooth-low-energy
1/16/2021, 6:26:47 PM - info - NestFactory: Starting Nest application...
1/16/2021, 6:26:48 PM - info - InstanceLoader: AppModule dependencies initialized
1/16/2021, 6:26:48 PM - info - InstanceLoader: ConfigModule dependencies initialized
1/16/2021, 6:26:48 PM - info - InstanceLoader: NestEmitterModule dependencies initialized
1/16/2021, 6:26:48 PM - info - InstanceLoader: IntegrationsModule dependencies initialized
1/16/2021, 6:26:48 PM - info - InstanceLoader: HttpModule dependencies initialized
1/16/2021, 6:26:48 PM - info - InstanceLoader: DiscoveryModule dependencies initialized
1/16/2021, 6:26:48 PM - info - InstanceLoader: ClusterModule dependencies initialized
1/16/2021, 6:26:48 PM - info - InstanceLoader: TerminusModule dependencies initialized
1/16/2021, 6:26:48 PM - info - InstanceLoader: BluetoothModule dependencies initialized
1/16/2021, 6:26:48 PM - info - InstanceLoader: ScheduleModule dependencies initialized
1/16/2021, 6:26:48 PM - info - InstanceLoader: EntitiesModule dependencies initialized
1/16/2021, 6:26:48 PM - info - InstanceLoader: HomeAssistantModule dependencies initialized
1/16/2021, 6:26:48 PM - info - InstanceLoader: BluetoothLowEnergyModule dependencies initialized
1/16/2021, 6:26:48 PM - info - InstanceLoader: StatusModule dependencies initialized
1/16/2021, 6:26:52 PM - info - RoutesResolver: EntitiesController {/entities}:
1/16/2021, 6:26:52 PM - info - RouterExplorer: Mapped {/entities, GET} route
1/16/2021, 6:26:52 PM - info - RoutesResolver: StatusController {/status}:
1/16/2021, 6:26:52 PM - info - RouterExplorer: Mapped {/status, GET} route
1/16/2021, 6:26:55 PM - info - HomeAssistantService: Successfully connected to MQTT broker at mqtt://192.168.11.2:1883
1/16/2021, 6:26:55 PM - info - ConfigService: Loading configuration from /opt/nodejs/lib/node_modules/room-assistant/dist/config/definitions/default.js, config/local.yml
1/16/2021, 6:26:56 PM - info - ClusterService: Starting mDNS advertisements and discovery
1/16/2021, 6:26:56 PM - info - NestApplication: Nest application successfully started
1/16/2021, 6:26:56 PM - info - HomeAssistantService: Device tracker requires manual setup in Home Assistant with topic: room-assistant/device_tracker/ble-426c7565436861726d426561636f6e73-3838-4949-tracker/state
1/16/2021, 6:26:56 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral 66dafe24f329 with ID 66dafe24f329 and RSSI -36
1/16/2021, 6:26:56 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral 50765cb7d9ea4e2199a4fa879613a492-42771-20348 with ID 50765cb7d9ea4e2199a4fa879613a492-42771-20348 and RSSI -70
1/16/2021, 6:26:56 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral 541379cd9fd0 with ID 541379cd9fd0 and RSSI -69
1/16/2021, 6:26:56 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral 43499ab9378e with ID 43499ab9378e and RSSI -57
1/16/2021, 6:26:56 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral 784f43a26bd6 with ID 784f43a26bd6 and RSSI -45
1/16/2021, 6:26:56 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral SoloAltusKeys6Pat with ID 426c7565436861726d426561636f6e73-3838-4954 and RSSI -40
1/16/2021, 6:26:56 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral SoloAltusKeys2Mini with ID 426c7565436861726d426561636f6e73-3838-4950 and RSSI -56
1/16/2021, 6:26:57 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral 1fed3a47c24c with ID 1fed3a47c24c and RSSI -72
1/16/2021, 6:26:57 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral 1f7dc3248008 with ID 1f7dc3248008 and RSSI -74
1/16/2021, 6:26:57 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral [TV] UN65JU6800 with ID fc8f909c0068 and RSSI -70
1/16/2021, 6:26:57 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral SoloAltusKeys1All with ID 426c7565436861726d426561636f6e73-3838-4949 and RSSI -56
1/16/2021, 6:26:57 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral 263433cce5a5 with ID 263433cce5a5 and RSSI -63
1/16/2021, 6:26:57 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral Nest Cam with ID 6416664b5d8c and RSSI -67
1/16/2021, 6:26:57 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral Nest Cam with ID 18b430e43188 and RSSI -68
1/16/2021, 6:26:57 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral 668a1cf1c01e with ID 668a1cf1c01e and RSSI -29
1/16/2021, 6:26:57 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral 29039d896bd9 with ID 29039d896bd9 and RSSI -62
1/16/2021, 6:26:57 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral 380195ca1bf6 with ID 380195ca1bf6 and RSSI -78
1/16/2021, 6:26:57 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral 42b5e9d23615 with ID 42b5e9d23615 and RSSI -50
1/16/2021, 6:26:57 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral 4c1c7d4445b1 with ID 4c1c7d4445b1 and RSSI -55
1/16/2021, 6:26:57 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral 40cbc0e808dc with ID 40cbc0e808dc and RSSI -55
1/16/2021, 6:26:57 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral soloAltusKeys3Par with ID 426c7565436861726d426561636f6e73-3838-4951 and RSSI -39
1/16/2021, 6:26:57 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral 426c7565436861726d426561636f6e73-3838-4953 with ID 426c7565436861726d426561636f6e73-3838-4953 and RSSI -33
1/16/2021, 6:26:58 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral 5e4d99ffca4b with ID 5e4d99ffca4b and RSSI -66
1/16/2021, 6:27:01 PM - error - ClusterService: dns service error: unknown
1/16/2021, 6:27:01 PM - error - ClusterService: dns service error: unknown
1/16/2021, 6:27:01 PM - error - ClusterService: dns service error: unknown
1/16/2021, 6:27:01 PM - error - ClusterService: dns service error: unknown
1/16/2021, 6:27:01 PM - error - ClusterService: dns service error: unknown
1/16/2021, 6:27:01 PM - error - ClusterService: dns service error: unknown
1/16/2021, 6:27:02 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral SMI-M1S with ID cc95839c2cd6 and RSSI -64
1/16/2021, 6:27:08 PM - info - ClusterService: Added 192.168.11.30:6425 to the cluster with id entry-room-assist-pi0w-6ab0ff-30
1/16/2021, 6:27:16 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral Eve with ID ceaf863bc24a and RSSI -85
1/16/2021, 6:27:36 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral 74d209239d5c with ID 74d209239d5c and RSSI -85
1/16/2021, 6:28:23 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral 18bb4e37a736 with ID 18bb4e37a736 and RSSI -54
1/16/2021, 6:28:23 PM - info - BluetoothLowEnergyService: Discovered new BLE peripheral 7119769d79a0 with ID 7119769d79a0 and RSSI -29
^Cpi@kitchen-room-assist-pi0w-131e32-31:~/room-assistant $ cat config/local.yml 
global:
  integrations:
    - homeAssistant
    - bluetoothLowEnergy
#
homeAssistant:
  mqttUrl: 'mqtt://192.168.11.2:1883'
  mqttOptions:
    username: mqttsabuca
    password: mqttpass
#
bluetoothLowEnergy:
  whitelist:
    - 42c67565436861726d426561636f6e73-3838-4950
    - 426c7565436861726d426561636f6e73-3838-4949
    - 42c67565436861726d426561636f6e73-3838-4951
    - 42c67565436861726d426561636f6e73-3838-4952
    - 42c67565436861726d426561636f6e73-3838-4953
    - 42c67565436861726d426561636f6e73-3838-4954
pi@kitchen-room-assist-pi0w-131e32-31:~/room-assistant $

**Comment by mKeRix** [comment]
I think that your whitelist may contain some typos - the discovered beacon IDs don't match your room-assistant configuration exactly (but they do seem to match your HA config!). Note that the working tag begins with `426c` (which matches the logs), while the others all begin with `42c6`. Could you try switching the `c` and the `6` around so that all begin with `426c` and see if that works out for you?

**Comment by bleitchhk** [comment]
Wow.  That fixed it and immediately.  All working perfectly !

Thanks for spotting the typo.  I did so many checks of the digits at the end of the UUID sequence, but not at front.

I am now going to try the OMRON integration.

Brendan
